docs/1-benchpark-setup.rst

```diff
@@ -0,0 +1,27 @@
+=========
+Benchpark setup
```

Comment:
I would like to propose the following change:

clone benchpark
cd benchpark --|
clone spack        |   --- these 4 lines can be bundled in 
clone ramble      |          "benchpark setup"
source files     --|

Then the user can be in $benchpark and do git pull or fork.
cd spack and do git pull or fork
cd ramble and do git pull or fork

Further changes are needed for the workspace to then look in $benchpark/spack and $benchpark/ramble for the files.  I'd like to avoid giving those paths as parameters to creating the workspace since I'd like to keep users from pointing those at the system spack of who knows what vintage.  Thoughts?

Reply:
> I would like to propose the following change:

Is the goal of this to avoid cloning ramble and Spack so much? I can see the benefit of that, as right now we do it for every benchmark/system combo we create (e.g. once for saxpy/openmp, once for saxpy/cuda, etc.). However:

* Each of these benchmark configs likely requires a full reinstall of all packages (e.g. the `+cuda` version of dependencies), which likely takes the bulk of the time
* The distinct-ramble-and-spack-for-each-benchmark approach allows users to make edits for one benchmark without affecting others (a possible advantage)
  * Admittedly this only applies for dependencies of the benchmark (i.e. we change package.py files in the package repo of a specific spack instance, and that doesn't affect the installation of other benchmark/system combos, but if we edit the `saxpy` package.py in `benchpark/repos`, then that *may* affect other benchmark installations)
  * Given the above point, the change you propose is IMO more consistent

I should note that even if you only want Spack/Ramble to be cloned once by benchpark, I can still have the script do that automatically (i.e. we don't have to tell users to do it manually).

> I'd like to avoid giving those paths as parameters to creating the workspace since I'd like to keep users from pointing those at the system spack of who knows what vintage. 

That's fine by me, although users might want to pull the latest benchpark and point it to an old Spack repo to make use of any reusable packages installed there, so I think they might want that in the future (I think that will be clearer though as we flesh out use cases).

Reply:
Ok, so I think from our discussion, lets do the following:
- The default behavior should be to clone out spack and ramble as ``benchpark setup`` does right now.  Please keep save their directories, so you can put the spack/ramble cache in those directories (i.e., /path/to/spack/.spack_cache) to keep their lifetimes to the lifetime of the clone.
- Optionally, ``benchpark setup`` should take a path to spack and/or ramble clone the user provides.  I think this enables the user to have a single instance of Spack/Ramble they can pass in for all of the experiments.  Since these may be system installs, I am not sure where to put the caches in this case?

@scheibelp could you make the above change to the python script, and point me at it, so i can make sure the docs match?